lib/internal/inspector_async_hook.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const { createHook } = require('async_hooks');
+const inspector = process.binding('inspector');
+const errors = require('internal/errors');
+
+if (!inspector || !inspector.asyncTaskScheduled) return;
+
+module.exports = createHook({
+  init(asyncId, type, triggerId, resource) {
+    if (!Number.isInteger(asyncId))
+      throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'asyncId', 'integer');
+    if (typeof type !== 'string')
+      throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'type', 'string');
```

Comment:
Yeah, sorry – I didn’t realize the binding methods were only called from other internal code.